File: app/src/androidTest/java/com/ezral/halo/MarketingBackdropActivity.java
package com.ezral.halo;

import android.app.Activity;
import android.os.Bundle;
import android.graphics.*;
import android.view.View;

/** Standalone test APK backdrop. Java avoids dependencies supplied only by the target APK. */
public class MarketingBackdropActivity extends Activity {
    @Override public void onCreate(Bundle state) {
        super.onCreate(state);
        getWindow().setStatusBarColor(Color.rgb(246,243,235));
        getWindow().setNavigationBarColor(Color.rgb(246,243,235));
        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR | View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR);
        setContentView(new View(this) {
            final Paint p = new Paint(Paint.ANTI_ALIAS_FLAG);
            void text(Canvas c, String t, float x, float y, float size, boolean bold) {
                p.setColor(Color.rgb(34,55,48)); p.setTextSize(size);
                p.setTypeface(Typeface.create("sans-serif", bold ? Typeface.BOLD : Typeface.NORMAL));
                c.drawText(t,x,y,p);
            }
            @Override protected void onDraw(Canvas c) {
                float s=getWidth()/400f; c.save(); c.scale(s,s); c.drawColor(Color.rgb(246,243,235));
                text(c,"S L O W   M O R N I N G S",28,40,11,true);
                p.setColor(Color.rgb(212,216,204)); c.drawRect(28,60,372,61,p);
                text(c,"A better",28,116,42,true); text(c,"morning cup.",28,164,42,true);
                text(c,"A simple pour-over ritual.",28,199,16,false);
                p.setColor(Color.rgb(222,227,205)); c.drawRoundRect(28,228,372,445,24,24,p);
                p.setColor(Color.rgb(163,173,143)); c.drawOval(102,385,300,420,p);
                p.setColor(Color.rgb(245,239,222)); c.drawRoundRect(267,286,319,359,24,24,p);
                p.setColor(Color.rgb(222,227,205)); c.drawRoundRect(277,300,307,346,13,13,p);
                p.setColor(Color.rgb(250,247,235)); c.drawRoundRect(125,280,282,397,36,36,p); c.drawOval(125,267,282,307,p);
                p.setColor(Color.rgb(112,73,49)); c.drawOval(139,279,268,303,p);
                text(c,"01",28,491,13,true); text(c,"Bloom",72,492,21,true);
                text(c,"Wet the grounds. Let the coffee open up.",72,519,13,false);
                text(c,"02",28,568,13,true); text(c,"Slow pour",72,569,21,true);
                text(c,"Pour gently, in small circles.",72,596,13,false);
                text(c,"03",28,645,13,true); text(c,"Enjoy",72,646,21,true);
                text(c,"Take your time. You have earned it.",72,673,13,false);
                text(c,"SAMPLE RECIPE  /  OVERLAY DEMONSTRATION",28,752,9,false);
                c.restore();
            }
        });
    }
}

File: app/src/androidTest/java/com/ezral/halo/MarketingBackdropActivity.kt
package com.ezral.halo

import android.app.Activity
import android.os.Bundle
import android.graphics.*
import android.view.View

/** Separate test-APK app, never shipped in Halo. Original recipe backdrop for honest overlay captures. */
class MarketingBackdropActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.statusBarColor = Color.rgb(246, 243, 235)
        window.navigationBarColor = Color.rgb(246, 243, 235)
        window.decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR or View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
        setContentView(object : View(this) {
            val p = Paint(Paint.ANTI_ALIAS_FLAG)
            override fun onDraw(c: Canvas) {
                val s = width / 400f
                c.save(); c.scale(s, s)
                c.drawColor(Color.rgb(246, 243, 235))
                fun text(t: String, x: Float, y: Float, size: Float, bold: Boolean = false, color: Int = Color.rgb(34, 55, 48)) {
                    p.color = color; p.textSize = size; p.typeface = Typeface.create("sans-serif", if (bold) Typeface.BOLD else Typeface.NORMAL)
                    c.drawText(t, x, y, p)
                }
                text("S L O W   M O R N I N G S", 28f, 40f, 11f, true)
                p.color = Color.rgb(212, 216, 204); c.drawRect(28f, 60f, 372f, 61f, p)
                text("A better", 28f, 116f, 42f, true)
                text("morning cup.", 28f, 164f, 42f, true)
                text("A simple pour-over ritual.", 28f, 199f, 16f)
                p.color = Color.rgb(222, 227, 205); c.drawRoundRect(28f, 228f, 372f, 445f, 24f, 24f, p)
                p.color = Color.rgb(163, 173, 143); c.drawOval(102f, 385f, 300f, 420f, p)
                p.color = Color.rgb(245, 239, 222); c.drawRoundRect(267f, 286f, 319f, 359f, 24f, 24f, p)
                p.color = Color.rgb(222, 227, 205); c.drawRoundRect(277f, 300f, 307f, 346f, 13f, 13f, p)
                p.color = Color.rgb(250, 247, 235); c.drawRoundRect(125f, 280f, 282f, 397f, 36f, 36f, p)
                c.drawOval(125f, 267f, 282f, 307f, p)
                p.color = Color.rgb(112, 73, 49); c.drawOval(139f, 279f, 268f, 303f, p)
                text("01", 28f, 491f, 13f, true); text("Bloom", 72f, 492f, 21f, true)
                text("Wet the grounds. Let the coffee open up.", 72f, 519f, 13f)
                text("02", 28f, 568f, 13f, true); text("Slow pour", 72f, 569f, 21f, true)
                text("Pour gently, in small circles.", 72f, 596f, 13f)
                text("03", 28f, 645f, 13f, true); text("Enjoy", 72f, 646f, 21f, true)
                text("Take your time. You have earned it.", 72f, 673f, 13f)
                text("SAMPLE RECIPE  /  OVERLAY DEMONSTRATION", 28f, 752f, 9f, false, Color.rgb(111, 123, 108))
                c.restore()
            }
        })
    }
}
